### PR TITLE
Cities Create

### PR DESCRIPTION
**Pull request by Light-it-luc**
# Light-it Pull Request Template ⚡⚡⚡

## Description ⭐

Implemented the "create" button of the Cities endpoint. Updated the routes and
CityController to handle the storing of new cities, after performing the neccessary
validation (through Form Request Validation, in the `StoreCityRequest` class).

Created a basic modal (without contents... the contents will be populated dynamically)
and used jQuery to send the POST request to the corresponding endpoint and update
the table without updating.

### Type of change 🏃

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [x] This change requires a documentation update

### Requires ❗

- [ ] This change requires a database update or migration
- [x] This change requires a documentation update

## Emoji Guide

**For reviewers: Emojis can be added to comments to call out blocking versus non-blocking feedback.**

E.g., Praise, minor suggestions, or clarifying questions that don’t block merging the PR.

> 🟢 Nice refactor!
> 🟡 Why was the default value removed?

E.g., Blocking feedback must be addressed before merging.

> 🔴 This change will break something important

|              |                |                                     |
| ------------ | -------------- | ----------------------------------- |
| Blocking     | 🔴 ❌ 🚨       | RED                                 |
| Non-blocking | 🟡 💡 🤔 💭    | Yellow, thinking, etc               |
| Praise       | 🟢 💚 😍 👍 🙌 | Green, hearts, positive emojis, etc |

## Links

- [Git Flow](https://lightit.slite.com/app/docs/SC8usN2Ju)
- [Handbook of good practices for reviewers in Code Reviews](https://lightit.slite.com/app/docs/ddNGohWthVB3fO)

### CODE REVIEW

File: resources/views/cities.blade.php [hunk 1]
```diff
@@ -72,5 +74,79 @@ class="del-btn ml-2 dark:bg-red-600 hover:bg-red-400">
...
+              $('[modal-title]').text(title)
+              $('[modal-content]').html(content)
+
+              modal.showModal()
+          }
+
+          function closeModal() {
+              var modal = document.getElementById('modal')
+              modal.close()
+          }
```
Comment (ezef) [inline resources/views/cities.blade.php]: no need to declare these functions inside `document.ready()` you can declare them outside.

File: resources/views/cities.blade.php [hunk 2]
```diff
@@ -72,5 +74,79 @@ class="del-btn ml-2 dark:bg-red-600 hover:bg-red-400">
...
+
+          $('[close-modal-btn]').click(closeModal)
+
+          $('button[create-button]').click(function() {
+              var city = {
```
Comment (ezef) [inline resources/views/cities.blade.php]: Always declare variables with let. https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Statements/let

But in this case you can declare it with `const city`

File: resources/views/cities.blade.php [hunk 3]
```diff
@@ -72,5 +74,79 @@ class="del-btn ml-2 dark:bg-red-600 hover:bg-red-400">
...
+                  name: $('input[name="name"]').val().trim(),
+                  country: $('input[name="country"]').val().trim()
+              }
+
+              $.ajax({
+                  type: 'POST',
+                  url: 'http://localhost:80/cities/create',
```
Comment (ezef) [inline resources/views/cities.blade.php]: no need to put absolute URL, you can use `/cities/create`

File: resources/views/cities.blade.php [hunk 4]
```diff
@@ -72,5 +74,79 @@ class="del-btn ml-2 dark:bg-red-600 hover:bg-red-400">
...
+                      )
+
+                      $('input[name="name"]').val('')
+                      $('input[name="country"]').val('')
+                  },
+                  error: function (err) {
+                    validationErrors = err.responseJSON.errors
+                    content = ''
+
+                    for(const failingField in validationErrors) {
```
Comment (ezef) [inline resources/views/cities.blade.php:139]: you can make use of map() to iterate https://developer.mozilla.org/es/docs/Web/JavaScript/Reference/Global_Objects/Array/map

File: app/Http/Requests/StoreCityRequest.php [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+     *
+     * @return array<string, \Illuminate\Contracts\Validation\ValidationRule|array<mixed>|string>
+     */
+    public function rules(Request $request): array
+    {
+        return [
+            'name' => ['required', 'max:255'],
+            'country' => [
+                'required', 'max:255',
+                Rule::unique('cities', 'country')
```
Comment (guillemartinicore) [inline app/Http/Requests/StoreCityRequest.php]: you should reconsider this

File: routes/web.php [hunk 6]
```diff
@@ -16,3 +16,4 @@
 */
 
 Route::get('cities', [CityController::class, 'index']);
+Route::post('cities/create', [CityController::class, 'store']);
```
Comment (guillemartinicore) [inline routes/web.php]: you can think of grouping routes with same prefixes. As you can see here, 'cities/' repeats many times here. I can give you an example via slack!

**Pull request merged**